autogen kube-daemon markdown files

mikedanese opened the pull request:
blame this mess on global variables. md autogen down. man to go.
@erictune @ingvagabund

mikedanese committed:
autogen daemon markdown

eparis commented on cmd/gendocs/gendocs.go:
```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
```
2014?

eparis commented:
Wow, this is hideous aweful and amazing all at once.  LGTM.

erictune commented on pkg/admission/plugins.go:
```diff
@@ -45,6 +46,7 @@ func GetPlugins() []string {
 	for k := range plugins {
 		keys = append(keys, k)
 	}
+	sort.Strings(keys)
```
Was this supposed to be in this PR?  Is so what does it do?  Make the generated output stable?

mikedanese replied:
Exactly. These keys are used to populate an apiserver arg.

erictune commented:
LGTM.

mikedanese commented:
@bgrant0607 @thockin can one of you add ok-to-merge

thockin commented on cmd/gendocs/hack/kubeapiserver/hack.go:
```diff
@@ -0,0 +1,34 @@
+		Use: "kube-apiserver",
+		Long: `The kubernetes API server validates and configures data for 3 types of objects: pods, services, and replicationcontrollers. Beyond just servicing REST operations, the API Server does two other things as well: 1. Schedules pods to worker nodes. Right now the scheduler is very simple. 2. Synchronize pod information (where they are, what ports they are exposing) with the service configurati ...
+
+The the kube-apiserver several options.`,
```
this sentence is gobbledegook

thockin commented on cmd/gendocs/gendocs.go:
```diff
@@ -0,0 +1,99 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
```
A comment would help sooooo much

thockin commented:
I don't really get what this is doing and whether this is a step towards a non-hack or some new perpetual hacks?

mikedanese commented:
@thockin I've added a comment to gendocs that hopefully explains this method of doc generation. I've also removed the word hack as I don't think this is necessarily a bad way to do this as it's very easily extensible to other cobra.Commands. The reason we are doing this is we have a couple of uses of the global `flag.CommandLine`, so we can't actually get an accurate view of the flags of multiple commands in a single process. Long term we could audit our usage of the global flagset and generate all this doc in a single process but for now I think this is ok.

mikedanese commented:
I'm moving this into v1-post milestone and porting doc changes to v1 in PR #10765

mikedanese commented:
I have a better way of doing this in the works that uses mungedoc macros.
